contracts/poap/src/msg.rs

```diff
@@ -0,0 +1,99 @@
+use cosmwasm_std::{Addr, Timestamp, Uint64};
+use cw721_base::InstantiateMsg as Cw721InstantiateMsg;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    /// Address of who will have the right to administer the contract.
+    pub admin: String,
+    /// Address of who can call the [`ExecuteMsg::MintTo`] other then the admin.
+    pub minter: String,
+    /// Id of the CW721 contract to initialize together with this contract.
+    pub cw721_code_id: Uint64,
+    /// Initialization message that will be sent to the CW721 contract.
+    pub cw721_initiate_msg: Cw721InstantiateMsg,
+    /// Information about the event.
+    pub event_info: EventInfo,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct EventInfo {
+    /// User that created the event.
+    pub creator: String,
+    /// Time at which the event begins.
+    pub start_time: Timestamp,
+    /// Time at which the event ends.
+    pub end_time: Timestamp,
+    /// Max amount of poap that a single user can mint.
+    pub per_address_limit: u32,
+    /// Identifies a valid IPFS URI corresponding to where the assets and metadata of the POAPs are stored.
+    pub base_poap_uri: String,
+    /// Uri of the poap event
+    pub event_uri: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    /// Allows the contract's admin to enable the [`ExecuteMsg::Mint`].
+    EnableMint {},
+    /// Allows the contract's admin to disable the [`ExecuteMsg::Mint`].
+    DisableMint {},
+    /// If the mint is enabled, allow the user to mint the poap by themself.
+    /// It's disabled after the event's end.
+    Mint {},
+    /// Allows the contract's admin or the minter to mint a POAP for a specific recipient.
+    /// It's disabled after the event's end.
+    MintTo { recipient: String },
+    /// Message that allows the event's creator to change the time frame of the event
+    /// if it's not started or finished.
+    UpdateEventInfo {
+        start_time: Timestamp,
+        end_time: Timestamp,
+    },
+    /// Allows the contract's admin to transfer the admin rights to another user.
+    UpdateAdmin { new_admin: String },
+    /// Allows the contract's admin to transfer the minting rights to another user.
+    UpdateMinter { new_minter: String },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    /// Returns the configuration info as a [`QueryConfigResponse`].
+    Config {},
+    /// Returns the event info as a [`QueryEventInfoResponse`].
+    EventInfo {},
+}
+
+/// Response to [`QueryMsg::Config`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct QueryConfigResponse {
```

Comment:
I think we can return `config` instead of separately fields.

Reply:
I don't like the idea of returing a struct used to store the contratc state, if in the future we need to extend that struct with some informations that are not required to queried from the client? Also this create an unecessary nesting that a client need to handle while deserializing and process the response

Reply:
Can you give me a sample the information that are not required to client? I think we should return all the config for the user who would get contract state.

Besides, we can reduce some code inside the query from [current one](https://github.com/desmos-labs/desmos-contracts/pull/35/files/03f02fdf42d61d27c22e63731da4d943667feae4#diff-2b9bf8d5b9ea44cca10b66d24526301c233a9e13e8c6dbc3debbb6c0b460b2d9R407-R413) to be like:
```rust
fn query_config(deps: Deps) -> StdResult<QueryConfigResponse> {
    let config = CONFIG.load(deps.storage)?;
    Ok(QueryConfigResponse { config })
}
```

Reply:
The `QueryMsg::EventInfo` don't returns the whole `EventInfo` struct stored inside the state but all the fields except `base_poap_uri`.  
Here the [ADR](https://github.com/desmos-labs/desmos-contracts/blob/master/docs/architecture/adr-001-poap-contract.md#eventinfo-1) reference.

Reply:
You are right. Making it separately has more flexibility. Let's remain the current structure.